got an exception when use encryt SM4 algorithm

proxy commit id : 51ec973a056b3414f80b2d207f2236ac184f6a2e

following is the step to reproduce the issue :
1. downlod the shardingsphere code and checkout as the corresponding commit version
2. compile the code and extract proxy distribution
3. use the following config and then start proxy

config-encrypt.yaml
```
schemaName: encrypt_db

dataSources:
  ds_0:
    url: jdbc:mysql://127.0.0.1:3306/encrypt_db?serverTimezone=UTC&useSSL=false
    username: root
    password:  
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 3000
    minPoolSize: 1

rules:
- !ENCRYPT
  encryptors:
    sm4_encryptor:
      type: SM4
      props:
        sm4-key: 123456ab
        sm4-mode: ECB
        sm4-padding: PKCS5Padding
    md5_encryptor:
      type: MD5
  tables:
    sbtest1:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest2:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest3:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest4:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest5:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest6:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest7:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest8:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest9:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
    sbtest10:
      columns:
        pad:
          cipherColumn: pad
          encryptorName: sm4_encryptor
  queryWithCipherColumn: true
```

server.yaml
```
rules:
  - !AUTHORITY
    users:
      - root@%:root
      - sharding@:sharding
    provider:
      type: ALL_PRIVILEGES_PERMITTED
#  - !TRANSACTION
#    defaultType: XA
#    providerType: Atomikos
#  - !SQL_PARSER
#    sqlCommentParseEnabled: true
#    sqlStatementCache:
#      initialCapacity: 2000
#      maximumSize: 65535
#      concurrencyLevel: 4
#    parseTreeCache:
#      initialCapacity: 128
#      maximumSize: 1024
#      concurrencyLevel: 4

props:
#  max-connections-size-per-query: 1
#  kernel-executor-size: 16  # Infinite by default.
#  proxy-frontend-flush-threshold: 128  # The default value is 128.
#  proxy-opentracing-enabled: false
#  proxy-hint-enabled: false
#  sql-show: false
#  check-table-metadata-enabled: false
#  show-process-list-enabled: false
#    # Proxy backend query fetch size. A larger value may increase the memory usage of ShardingSphere Proxy.
#    # The default value is -1, which means set the minimum value for different JDBC drivers.
#  proxy-backend-query-fetch-size: -1
#  check-duplicate-table-enabled: false
#  proxy-frontend-executor-size: 0 # Proxy frontend executor size. The default value is 0, which means let Netty decide.
#    # Available options of proxy backend executor suitable: OLAP(default), OLTP. The OLTP option may reduce time cost of writing packets to client, but it may increase the latency of SQL execution
#    # if client connections are more than proxy-frontend-netty-executor-size, especially executing slow SQL.
  proxy-backend-executor-suitable: OLTP
#  proxy-frontend-max-connections: 0 # Less than or equal to 0 means no limitation.
#  sql-federation-enabled: false
#    # Available proxy backend driver type: JDBC (default), ExperimentalVertx
#  proxy-backend-driver-type: JDBC
```

and then modify the sysbench oltp_common.lua, change the create table clause, make pad to char(200)
start sysbench to test
```
sysbench oltp_read_only --mysql-host=127.0.0.1 --mysql-port=3307 --mysql-user=root --mysql-password='root' --mysql-db=encrypt_db --tables=10 --table-size=1000000 --report-interval=10 --time=3600 --threads=10 --max-requests=0 --percentile=99 --mysql-ignore-errors="all" --rand-type=uniform --range_selects=off --auto_inc=off cleanup
sysbench oltp_read_only --mysql-host=127.0.0.1 --mysql-port=3307 --mysql-user=root --mysql-password='root' --mysql-db=encrypt_db --tables=10 --table-size=1000000 --report-interval=10 --time=3600 --threads=10 --max-requests=0 --percentile=99 --mysql-ignore-errors="all" --rand-type=uniform --range_selects=off --auto_inc=off prepare

sysbench oltp_point_select --mysql-host=127.0.0.1 --mysql-port=3307 --mysql-user=root --mysql-password='root' --mysql-db=encrypt_db --tables=10 --table-size=1000000 --report-interval=5 --time=300 --threads=64 --max-requests=0 --percentile=99 --mysql-ignore-errors="all" --range_selects=off --rand-type=uniform --auto_inc=off run | tee oltp_point_select_64.txt

```

and you will get the following problem :

```
[ERROR] 2022-01-13 16:21:43.226 [epollEventLoopGroup-3-7] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
java.lang.IllegalStateException: Mode must be either CBC or ECB.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:508)
	at org.apache.shardingsphere.encrypt.algorithm.SM4EncryptAlgorithm.checkAndGetMode(SM4EncryptAlgorithm.java:123)
	at org.apache.shardingsphere.encrypt.algorithm.SM4EncryptAlgorithm.sm4(SM4EncryptAlgorithm.java:103)
	at org.apache.shardingsphere.encrypt.algorithm.SM4EncryptAlgorithm.encrypt(SM4EncryptAlgorithm.java:76)
	at org.apache.shardingsphere.encrypt.algorithm.SM4EncryptAlgorithm.encrypt(SM4EncryptAlgorithm.java:72)
	at org.apache.shardingsphere.encrypt.algorithm.SM4EncryptAlgorithm.encrypt(SM4EncryptAlgorithm.java:40)
	at org.apache.shardingsphere.encrypt.rewrite.token.generator.impl.EncryptInsertValuesTokenGenerator.setCipherColumn(EncryptInsertValuesTokenGenerator.java:170)
	at org.apache.shardingsphere.encrypt.rewrite.token.generator.impl.EncryptInsertValuesTokenGenerator.encryptToken(EncryptInsertValuesTokenGenerator.java:128)
	at org.apache.shardingsphere.encrypt.rewrite.token.generator.impl.EncryptInsertValuesTokenGenerator.generateNewSQLToken(EncryptInsertValuesTokenGenerator.java:94)
	at org.apache.shardingsphere.encrypt.rewrite.token.generator.impl.EncryptInsertValuesTokenGenerator.generateSQLToken(EncryptInsertValuesTokenGenerator.java:67)
	at org.apache.shardingsphere.encrypt.rewrite.token.generator.impl.EncryptInsertValuesTokenGenerator.generateSQLToken(EncryptInsertValuesTokenGenerator.java:50)
	at org.apache.shardingsphere.infra.rewrite.sql.token.generator.SQLTokenGenerators.generateSQLTokens(SQLTokenGenerators.java:70)
	at org.apache.shardingsphere.infra.rewrite.context.SQLRewriteContext.generateSQLTokens(SQLRewriteContext.java:83)
	at org.apache.shardingsphere.infra.rewrite.SQLRewriteEntry.createSQLRewriteContext(SQLRewriteEntry.java:78)
	at org.apache.shardingsphere.infra.rewrite.SQLRewriteEntry.rewrite(SQLRewriteEntry.java:70)
	at org.apache.shardingsphere.infra.context.kernel.KernelProcessor.rewrite(KernelProcessor.java:59)
	at org.apache.shardingsphere.infra.context.kernel.KernelProcessor.generateExecutionContext(KernelProcessor.java:47)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:115)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:67)
	at org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandler.execute(SchemaAssignedDatabaseBackendHandler.java:55)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:62)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:96)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:69)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/SM3EncryptAlgorithm.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/SM3EncryptAlgorithm.java
@@ -18,11 +18,12 @@
 package org.apache.shardingsphere.encrypt.algorithm;
 
 import com.google.common.base.Preconditions;
+import lombok.Getter;
 import lombok.Setter;
-import org.apache.commons.codec.binary.Hex;
 import org.apache.commons.codec.binary.StringUtils;
 import org.apache.shardingsphere.encrypt.spi.EncryptAlgorithm;
 import org.bouncycastle.crypto.digests.SM3Digest;
+import org.bouncycastle.pqc.math.linearalgebra.ByteUtils;
 
 import java.security.Security;
 import java.util.Properties;
@@ -30,39 +31,44 @@ import java.util.Properties;
 /**
  * SM3 encrypt algorithm.
  */
+@Getter
+@Setter
 public final class SM3EncryptAlgorithm implements EncryptAlgorithm<Object, String> {
     
     static {
         Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
     }
     
-    private static final String SM3 = "SM3";
-    
     private static final String SM3_SALT = "sm3-salt";
     
     private static final int SALT_LENGTH = 8;
     
-    @Setter
-    private Properties props;
-
+    private Properties props = new Properties();
+    
+    private byte[] sm3Salt;
+    
+    @Override
+    public void init() {
+        sm3Salt = createSm3Salt();
+    }
+    
+    private byte[] createSm3Salt() {
+        String salt = null == props.getProperty(SM3_SALT) ? "" : String.valueOf(props.getProperty(SM3_SALT));
+        Preconditions.checkState(0 == salt.length() || SALT_LENGTH == salt.length(), "Salt should be either blank or better " + SALT_LENGTH + " bytes long.");
+        return 0 == salt.length() ? new byte[0] : StringUtils.getBytesUtf8(salt);
+    }
+    
     @Override
     public String encrypt(final Object plainValue) {
         if (null == plainValue) {
             return null;
         }
-        String salt = null == props.getProperty(SM3_SALT) ? "" : String.valueOf(props.getProperty(SM3_SALT));
-        Preconditions.checkState(0 == salt.length() || SALT_LENGTH == salt.length(), "Salt should be either blank or better " + SALT_LENGTH + " bytes long.");
-        return Hex.encodeHexString(digest(StringUtils.getBytesUtf8(String.valueOf(plainValue)), 0 == salt.length() ? new byte[0] : StringUtils.getBytesUtf8(salt)));
+        return ByteUtils.toHexString(digest(StringUtils.getBytesUtf8(String.valueOf(plainValue)), sm3Salt));
     }
     
     @Override
     public Object decrypt(final String cipherValue) {
         return cipherValue;
-    }
-    
-    @Override
-    public String getType() {
-        return SM3;
     }
     
     private byte[] digest(final byte[] input, final byte[] salt) {
@@ -84,6 +90,7 @@ public final class SM3EncryptAlgorithm implements EncryptAlgorithm<Object, Strin
     }
     
     @Override
-    public void init() {
+    public String getType() {
+        return "SM3";
     }
 }

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/SM4EncryptAlgorithm.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/SM4EncryptAlgorithm.java
@@ -18,12 +18,13 @@
 package org.apache.shardingsphere.encrypt.algorithm;
 
 import com.google.common.base.Preconditions;
+import lombok.Getter;
 import lombok.Setter;
 import lombok.SneakyThrows;
-import org.apache.commons.codec.binary.Hex;
 import org.apache.commons.codec.binary.StringUtils;
 import org.apache.shardingsphere.encrypt.spi.EncryptAlgorithm;
-import org.apache.shardingsphere.infra.exception.ShardingSphereException;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.pqc.math.linearalgebra.ByteUtils;
 
 import javax.crypto.Cipher;
 import javax.crypto.spec.IvParameterSpec;
@@ -31,12 +32,15 @@ import javax.crypto.spec.SecretKeySpec;
 import java.security.Security;
 import java.util.Arrays;
 import java.util.HashSet;
+import java.util.Optional;
 import java.util.Properties;
 import java.util.Set;
 
 /**
  * SM4 encrypt algorithm.
  */
+@Getter
+@Setter
 public final class SM4EncryptAlgorithm implements EncryptAlgorithm<Object, String> {
     
     static {
@@ -61,19 +65,66 @@ public final class SM4EncryptAlgorithm implements EncryptAlgorithm<Object, Strin
     
     private static final Set<String> PADDINGS = new HashSet<>(Arrays.asList("PKCS5Padding", "PKCS7Padding"));
     
-    @Setter
-    private Properties props;
+    private Properties props = new Properties();
+    
+    private String sm4Mode;
+    
+    private byte[] sm4Key;
+    
+    private byte[] sm4Iv;
+    
+    private String sm4Padding;
+    
+    @Override
+    public void init() {
+        sm4Mode = createSm4Mode();
+        sm4Key = createSm4Key();
+        sm4Iv = createSm4Iv(sm4Mode);
+        sm4Padding = createSm4Padding();
+    }
+    
+    private String createSm4Mode() {
+        Preconditions.checkArgument(props.containsKey(SM4_MODE), "%s can not be null.", SM4_MODE);
+        String result = String.valueOf(props.getProperty(SM4_MODE)).toUpperCase();
+        Preconditions.checkState(MODES.contains(result), "Mode must be either CBC or ECB.");
+        return result;
+    }
+    
+    private byte[] createSm4Key() {
+        Preconditions.checkArgument(props.containsKey(SM4_KEY), "%s can not be null.", SM4_KEY);
+        String sm4KeyValue = String.valueOf(props.getProperty(SM4_KEY));
+        byte[] sm4KeyBytes = ByteUtils.fromHexString(sm4KeyValue);
+        Preconditions.checkState(KEY_LENGTH == sm4KeyBytes.length, "Key length must be " + KEY_LENGTH + " bytes long.");
+        return sm4KeyBytes;
+    }
+    
+    private byte[] createSm4Iv(final String sm4Mode) {
+        if ("CBC".equalsIgnoreCase(sm4Mode)) {
+            Preconditions.checkArgument(props.containsKey(SM4_IV), "%s can not be null.", SM4_IV);
+            String sm4IvValue = String.valueOf(props.getProperty(SM4_IV));
+            byte[] sm4IvBytes = ByteUtils.fromHexString(sm4IvValue);
+            Preconditions.checkState(IV_LENGTH == sm4IvBytes.length, "Iv length must be " + IV_LENGTH + " bytes long.");
+        }
+        return null;
+    }
+    
+    private String createSm4Padding() {
+        Preconditions.checkArgument(props.containsKey(SM4_PADDING), "%s can not be null.", SM4_PADDING);
+        String result = String.valueOf(props.get(SM4_PADDING)).toUpperCase().replace("PADDING", "Padding");
+        Preconditions.checkState(PADDINGS.contains(result), "Padding must be either PKCS5Padding or PKCS7Padding.");
+        return result;
+    }
     
     @Override
     public String encrypt(final Object plainValue) {
         if (null == plainValue) {
             return null;
         }
-        return Hex.encodeHexString(encrypt(StringUtils.getBytesUtf8(String.valueOf(plainValue))));
+        return ByteUtils.toHexString(encrypt(StringUtils.getBytesUtf8(String.valueOf(plainValue))));
     }
     
     private byte[] encrypt(final byte[] plainValue) {
-        return sm4(plainValue, Cipher.ENCRYPT_MODE);
+        return handle(plainValue, Cipher.ENCRYPT_MODE);
     }
     
     @SneakyThrows
@@ -82,60 +133,32 @@ public final class SM4EncryptAlgorithm implements EncryptAlgorithm<Object, Strin
         if (null == cipherValue) {
             return null;
         }
-        return StringUtils.newStringUtf8(decrypt(Hex.decodeHex(cipherValue.toCharArray())));
+        return StringUtils.newStringUtf8(decrypt(ByteUtils.fromHexString(cipherValue)));
     }
     
     private byte[] decrypt(final byte[] cipherValue) {
-        return sm4(cipherValue, Cipher.DECRYPT_MODE);
-    }
-    
-    @Override
-    public void init() {
-    }
-    
-    @Override
-    public String getType() {
-        return SM4;
+        return handle(cipherValue, Cipher.DECRYPT_MODE);
     }
 
     @SneakyThrows
-    private byte[] sm4(final byte[] input, final int mode) {
-        String modeAndPadding = String.format("SM4/%s/%s", checkAndGetMode(), checkAndGetPadding());
-        Cipher cipher = Cipher.getInstance(modeAndPadding, org.bouncycastle.jce.provider.BouncyCastleProvider.PROVIDER_NAME);
-        SecretKeySpec secretKeySpec = new SecretKeySpec(Hex.decodeHex(checkAndGetKey().toCharArray()), SM4);
-        String iv = checkAndGetIv(modeAndPadding);
-        if (null != iv) {
-            cipher.init(mode, secretKeySpec, new IvParameterSpec(Hex.decodeHex(iv.toCharArray())));
+    private byte[] handle(final byte[] input, final int mode) {
+        Cipher cipher = Cipher.getInstance(String.format("SM4/%s/%s", sm4Mode, sm4Padding), BouncyCastleProvider.PROVIDER_NAME);
+        SecretKeySpec secretKeySpec = new SecretKeySpec(sm4Key, SM4);
+        Optional<byte[]> sm4Iv = getSm4Iv();
+        if (sm4Iv.isPresent()) {
+            cipher.init(mode, secretKeySpec, new IvParameterSpec(sm4Iv.get()));
         } else {
             cipher.init(mode, secretKeySpec);
         }
         return cipher.doFinal(input);
     }
     
-    private String checkAndGetKey() throws ShardingSphereException {
-        String result = null == props.getProperty(SM4_KEY) ? null : String.valueOf(props.getProperty(SM4_KEY));
-        Preconditions.checkState(KEY_LENGTH != result.length(), "Key length must be " + KEY_LENGTH + " bytes long.");
-        return result;
+    private Optional<byte[]> getSm4Iv() {
+        return Optional.ofNullable(sm4Iv);
     }
     
-    private String checkAndGetMode() throws ShardingSphereException {
-        String result = null == props.getProperty(SM4_MODE) ? null : String.valueOf(props.getProperty(SM4_MODE)).toUpperCase();
-        Preconditions.checkState(MODES.contains(result), "Mode must be either CBC or ECB.");
-        return result;
-    }
-    
-    private String checkAndGetPadding() throws ShardingSphereException {
-        Object objectPadding = props.getProperty(SM4_PADDING);
-        String result = null == objectPadding ? null : String.valueOf(objectPadding).toUpperCase().replace("PADDING", "Padding");
-        Preconditions.checkState(PADDINGS.contains(result), "Padding must be either PKCS5Padding or PKCS7Padding.");
-        return result;
-    }
-    
-    private String checkAndGetIv(final String mode) throws ShardingSphereException {
-        String result = null == props.getProperty(SM4_IV) ? null : String.valueOf(props.getProperty(SM4_IV));
-        if ("CBC".equalsIgnoreCase(mode)) {
-            Preconditions.checkState(null == result || IV_LENGTH != result.length(), "Iv length must be " + IV_LENGTH + " bytes long.");
-        }
-        return result;
+    @Override
+    public String getType() {
+        return SM4;
     }
 }

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/SM4EncryptAlgorithm.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/SM4EncryptAlgorithm.java
@@ -67,20 +67,19 @@ public final class SM4EncryptAlgorithm implements EncryptAlgorithm<Object, Strin
     
     private Properties props = new Properties();
     
-    private String sm4Mode;
-    
     private byte[] sm4Key;
     
     private byte[] sm4Iv;
     
-    private String sm4Padding;
+    private String sm4ModePadding;
     
     @Override
     public void init() {
-        sm4Mode = createSm4Mode();
+        String sm4Mode = createSm4Mode();
+        String sm4Padding = createSm4Padding();
+        sm4ModePadding = "SM4/" + sm4Mode + "/" + sm4Padding;
         sm4Key = createSm4Key();
         sm4Iv = createSm4Iv(sm4Mode);
-        sm4Padding = createSm4Padding();
     }
     
     private String createSm4Mode() {
@@ -142,7 +141,7 @@ public final class SM4EncryptAlgorithm implements EncryptAlgorithm<Object, Strin
 
     @SneakyThrows
     private byte[] handle(final byte[] input, final int mode) {
-        Cipher cipher = Cipher.getInstance(String.format("SM4/%s/%s", sm4Mode, sm4Padding), BouncyCastleProvider.PROVIDER_NAME);
+        Cipher cipher = Cipher.getInstance(sm4ModePadding, BouncyCastleProvider.PROVIDER_NAME);
         SecretKeySpec secretKeySpec = new SecretKeySpec(sm4Key, SM4);
         Optional<byte[]> sm4Iv = getSm4Iv();
         if (sm4Iv.isPresent()) {

--- a/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/algorithm/SM4EncryptAlgorithmTest.java
+++ b/shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/test/java/org/apache/shardingsphere/encrypt/algorithm/SM4EncryptAlgorithmTest.java
@@ -52,7 +52,7 @@ public final class SM4EncryptAlgorithmTest {
         assertThat(encryptAlgorithm.encrypt("test"), is("028654f2ca4f575dee9e1faae85dadde"));
     }
     
-    @Test(expected = NullPointerException.class)
+    @Test(expected = IllegalArgumentException.class)
     public void assertEncryptWithoutKey() {
         Properties props = new Properties();
         encryptAlgorithm.setProps(props);
@@ -72,7 +72,7 @@ public final class SM4EncryptAlgorithmTest {
         assertThat(encryptAlgorithm.decrypt("028654f2ca4f575dee9e1faae85dadde").toString(), is("test"));
     }
     
-    @Test(expected = NullPointerException.class)
+    @Test(expected = IllegalArgumentException.class)
     public void assertDecryptWithoutKey() {
         Properties props = new Properties();
         encryptAlgorithm.setProps(props);

